src/burn/engine/variable.cpp

```diff
@@ -1961,6 +1963,56 @@ static HRESULT InitializeVariableRebootPending(
         }
     }
 
+    if (!fRebootPending)
+    {
+        hr = RegKeyReadNumber(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Microsoft\\ServerManager", L"CurrentRebootAttempts", TRUE, &dwValue);
+        fRebootPending = SUCCEEDED(hr) && 0 < dwValue;
+
+        if (!fRebootPending)
+        {
+            hr = RegKeyReadNumber(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Microsoft\\Updates", L"UpdateExeVolatile", TRUE, &dwValue);
+            fRebootPending = SUCCEEDED(hr) && 0 < dwValue;
+
+            if (!fRebootPending)
+            {
+                fRebootPending = RegValueExists(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Component Based Servicing\\RebootPending", NULL, TRUE);
+
+                if (!fRebootPending)
+                {
+                    fRebootPending = RegValueExists(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Component Based Servicing\\RebootInProgress", NULL, TRUE);
+
+                    if (!fRebootPending)
+                    {
+                        hr = RegKeyReadNumber(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\WindowsUpdate\\Auto Update", L"AUState", TRUE, &dwValue);
+                        fRebootPending = SUCCEEDED(hr) && 8 == dwValue;
+
+                        if (!fRebootPending)
+                        {
+                            fRebootPending = RegValueExists(HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Control\\Session Manager", L"PendingFileRenameOperations", TRUE);
+
+                            if (!fRebootPending)
+                            {
+                                fRebootPending = RegValueExists(HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Control\\Session Manager", L"PendingFileRenameOperations2", TRUE);
+
+                                if (!fRebootPending)
```

Comment:
The approach of so many four-line blocks:

    if (!fRebootPending)
    {
        ExitFunction();
    }

over and over made me die a little inside. I could invent a new ExitOnFalse macro...